src/main/java/de/bixilon/minosoft/gui/rendering/gui/gui/screen/menu/options/GUISettingsMenu.kt

```diff
@@ -0,0 +1,130 @@
+
+        wawlaEnabledButton = ButtonElement(guiRenderer, formatEnabled("menu.options.gui.wawla", wawlaProfile.enabled)) {
+            wawlaProfile.enabled = !wawlaProfile.enabled
+            wawlaEnabledButton.textElement.text = formatEnabled("menu.options.gui.wawla", wawlaProfile.enabled)
+            updateDisabledStates()
+        }
+        this += wawlaEnabledButton
+
+        wawlaLimitReachButton = ButtonElement(guiRenderer, formatEnabled("menu.options.gui.wawla_limit_reach", wawlaProfile.limitReach)) {
+            wawlaProfile.limitReach = !wawlaProfile.limitReach
+            wawlaLimitReachButton.textElement.text = formatEnabled("menu.options.gui.wawla_limit_reach", wawlaProfile.limitReach)
+        }
+        this += wawlaLimitReachButton
+
+        wawlaIdentifierButton = ButtonElement(guiRenderer, formatEnabled("menu.options.gui.wawla_identifier", wawlaProfile.identifier)) {
+            wawlaProfile.identifier = !wawlaProfile.identifier
+            wawlaIdentifierButton.textElement.text = formatEnabled("menu.options.gui.wawla_identifier", wawlaProfile.identifier)
+        }
+        this += wawlaIdentifierButton
+
+        wawlaBlockEnabledButton = ButtonElement(guiRenderer, formatEnabled("menu.options.gui.wawla_block", wawlaProfile.block.enabled)) {
+            wawlaProfile.block.enabled = !wawlaProfile.block.enabled
+            wawlaBlockEnabledButton.textElement.text = formatEnabled("menu.options.gui.wawla_block", wawlaProfile.block.enabled)
+        }
+        this += wawlaBlockEnabledButton
+
+        wawlaEntityEnabledButton = ButtonElement(guiRenderer, formatEnabled("menu.options.gui.wawla_entity", wawlaProfile.entity.enabled)) {
+            wawlaProfile.entity.enabled = !wawlaProfile.entity.enabled
+            wawlaEntityEnabledButton.textElement.text = formatEnabled("menu.options.gui.wawla_entity", wawlaProfile.entity.enabled)
+        }
+        this += wawlaEntityEnabledButton
+
+        wawlaEntityHealthButton = ButtonElement(guiRenderer, formatEnabled("menu.options.gui.wawla_entity_health", wawlaProfile.entity.health)) {
+            wawlaProfile.entity.health = !wawlaProfile.entity.health
+            wawlaEntityHealthButton.textElement.text = formatEnabled("menu.options.gui.wawla_entity_health", wawlaProfile.entity.health)
+        }
+        this += wawlaEntityHealthButton
+
+        wawlaEntityHandButton = ButtonElement(guiRenderer, formatEnabled("menu.options.gui.wawla_entity_hand", wawlaProfile.entity.hand)) {
+            wawlaProfile.entity.hand = !wawlaProfile.entity.hand
+            wawlaEntityHandButton.textElement.text = formatEnabled("menu.options.gui.wawla_entity_hand", wawlaProfile.entity.hand)
+        }
+        this += wawlaEntityHandButton
+
+        this += SpacerElement(guiRenderer, Vec2f(0f, 10f))
+        this += ButtonElement(guiRenderer, "menu.options.done".i18n()) { guiRenderer.gui.pop() }
+
+        updateDisabledStates()
+    }
+
+    private fun updateDisabledStates() {
```

Comment:
The profile provides delegates. You can easily observe it. Justs create a profile synced button (or so), that observes the profile and updates the button when that "event" is triggered. No need for polling.